src/robot/libraries/Remote.py

```diff
@@ -227,23 +228,49 @@ def _server(self):
         finally:
             server('close')()
 
+    @property
+    def _kw_cache(self):
+        """
+        returns False or a dict with structure:
+        { "keyword_name" : { 'args':[], 'tags':[], 'doc':"", 'types':[]},
+          "kw2"          : {...}, etc. }
+        """
+        if self.__kw_cache is None:
+            self.__kw_cache = False
+            try:
+                with self._server as server:
+                    self.__kw_cache = server.get_library_information()
+            except:
+                pass # Best efort failed, fall back to regular loading
+        return self.__kw_cache
+
     def get_keyword_names(self):
+        if self._kw_cache:
```

Comment:
Not sure what is possible with the Python version we need to support but I would try to implement decorator similiar to lru_cache instead, with optional argument. Ie:

@keyword_cache()

@keyword_cache(type='tags')
..

Reply:
I have now also processed this comment.